website/src/content/docs/docs/howtos/ARM/versioning.md

```diff
@@ -0,0 +1,315 @@
+
+**Example: Adding an operation in a later version**
+
+```tsp
+@armResourceOperations
+interface Employees {
+  get is ArmResourceRead<Employee>;
+  // v3: Add createOrUpdate operation
+  @added(Versions.v3)
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+}
+```
+
+### @removed
+
+To remove models, properties, or operations starting from a specific version, use the `@removed` decorator.
+
+- The `version` argument is the version where the element is removed.
+- The element will not be present in that version or any later versions.
+
+**Example: Removing a property and a model across versions**
+
+For example:
+
+```tsp
+// v1: Employee model with 'city' property
+model Employee {
+  name: string;
+  city?: string;
+}
+
+// v2: Remove 'city' property
+model Employee {
+  name: string;
+
+  @removed(Versions.v2)
+  city?: string;
+}
+
+// v3: Remove Employee model entirely
+@removed(Versions.v3)
+model Employee {
+  name: string;
+}
+```
+
+**Example: Removing an operation**
+
+```tsp
+@armResourceOperations
+interface Employees {
+  get is ArmResourceRead<Employee>;
+  @removed(Versions.v3)
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+}
+```
+
+### @madeRequired and @madeOptional
+
+Use these decorators to change whether a property is required or optional in a specific version.
+
+- Use `@madeOptional(Versions.version)` to make a property optional starting in that version.
+- Use `@madeRequired(Versions.version)` to make a property required starting in that version.
+
+**Example: Changing a property from required to optional, then back to required**
+
+For example:
+
+```tsp
+// v1: movingStatus is required
+model MoveResponse {
+  movingStatus: string;
+}
+
+// v2: movingStatus becomes optional
+model MoveResponse {
+  @madeOptional(Versions.v2)
+  movingStatus?: string;
+}
+```
+
+### @renamedFrom
+
+Use `@renamedFrom` to rename models, properties, operations, enums, etc., in a specific version.
+
+- The `version` argument is the version where the name changed.
+- The `oldName` argument is the previous name.
+
+**Example: Renaming a property and a model across versions**
+
+For example:
+
+```tsp
+// v1: Model and property have original names
+model WorkerProperties {
+  state?: string;
+}
+
+// v2: Rename property 'state' to 'city'
+model WorkerProperties {
+  @renamedFrom(Versions.v2, "state")
+  city?: string;
+}
+
+// v3: Rename model 'WorkerProperties' to 'EmployeeProperties'
+@renamedFrom(Versions.v3, "WorkerProperties")
+model EmployeeProperties {
+  @renamedFrom(Versions.v2, "state")
+  city?: string;
+}
+```
+
+### @returnTypeChangedFrom
+
+Use this decorator to change the return type of an operation in a specific version.
+
+```tsp
+@armResourceOperations
+interface Employees {
+  @removed(Versions.v2)
+  @returnTypeChangedFrom(Versions.v2, Worker)
+  get is ArmResourceRead<Employee>;
+}
+
+model Employee {
+  name: string;
+}
+
+model Worker {
+  name: string;
+}
+```
+
+## Complex Scenarios
```

Comment:
I think this should also contain the simple scenarios from the 'Getting Started'